src/Numerics/Interpolation/CubicSpline.cs

```diff
@@ -623,5 +623,87 @@ int LeftSegmentIndex(double t)
 
             return Math.Min(Math.Max(index, 0), _x.Length - 2);
         }
+
+        /// <summary>
+        /// Gets all the t values where the derivative is 0
+        /// </summary>
+        /// <returns>An array of t values (in the domain of teh function) where the derivative of the spline is 0</returns>
+        public double[] GetHorizontalDerivativeTValues()
+        {
+            List<double> points = new List<double>();
+            for (int index = 0; index < _x.Length - 1; index++)
+            {
+                double a = 6 * _c3[index]; //derive ax^3 and multiply by 2
+                double b = 2 * _c2[index]; //derive bx^2
+                double c = _c1[index];//derive cx
+                double d = b * b - 2 * a * c;
+                //first check if a is 0, if so its a linear function, this happens with quadratic condition
+                if (a.AlmostEqual(0))
+                {
+                    double x = _x[index] - c / b;
+                    //check if the result is in the domain
+                    if (_x[index] <= x && x <= _x[index + 1]) points.Add(x);
+                }
+                else if (d.AlmostEqual(0))//its a quadratic with a single solution
+                {
+                    double x = _x[index] - b / a;
+                    if (_x[index] <= x && x <= _x[index + 1]) points.Add(x);
+                }
+                else if (d > 0)//only has a solution if d is greater than 0
+                {
+                    d = (double)System.Math.Sqrt(d);
+                    //apply quadratic equation
+                    double x1 = _x[index] + (-b + d) / a;
+                    double x2 = _x[index] + (-b - d) / a;
+                    //Add any solution points that fall within the domain to the list
+                    if ((_x[index] <= x1) && (x1 <= _x[index + 1])) points.Add(x1);
+                    if ((_x[index] <= x2) && (x2 <= _x[index + 1])) points.Add(x2);
+                }
+            }
+            return points.ToArray();
+        }
+
+        /// <summary>
+        /// Returns the t values in the domain of the spline for which it takes the minimum and maximum value.
+        /// </summary>
+        /// <returns>A tuple containing the t value for which the spline is minimum in the first component and maximum in the second component </returns>
+        public Tuple<double, double> GetMinMaxTValues()
```

Comment:
The extrema of a differentiable function on a closed interval will be at its critical points or at the edges of the interval. Not sure the each input (x, y) point has to be tested.

For eg, something like this should be sufficient to get the co-ord of the minimum:
```csharp
var xs = CriticalPoints().Append(_x[0]).Append(_x[^1]);
var ys = xs.Select(x => Interpolate(x));
var minCoord = xs.Zip(ys, (x, y) => (x, y)).MinBy(p => p.y);
```

Given how simple the above code is, maybe this function might not be necessary?
If you do decide to keep it, then `Extrema()` might be a better name.